# Iseng2 bikin Vaksin Project - DENV

_Muhammad Adnan_<sup>1</sup>, _Matin Nuhamunada_<sup>1*</sup>, and _Lisna Hidayat_<sup>1</sup>

<sup>1</sup>Department of Tropical Biology, Universitas Gadjah Mada;   
Jl. Teknika Selatan, Sekip Utara, Bulaksumur, Yogyakarta, Indonesia, 55281;   

*Correspondence: [matin_nuhamunada@ugm.ac.id](mailto:matin_nuhamunada@mail.ugm.ac.id)  

---
## Abstract
Abstract here

### Keywords
Keywords here


In [130]:
from Bio import SeqIO
import pandas as pd
import bs4
from bs4 import BeautifulSoup

# Load UNIPROT Multiple Fasta File

Q: Jelaskan strategi dalam mencari dan mengumpulkan sekuens protein yang anda pilih sebagai kandidat antigen!

A: organism "dengue virus 2", gene name "e", protein name " envelope glycoprotein"

In [131]:
sequence =  list(SeqIO.parse("UNIPROT_Dengue Virus Type 2_Envelope Glycoprotein_Gene E_Virus 2.fasta", "fasta"))
print(len(sequence))

232


# Analyze VaxiJen Result

Q: VaxiJen digunakan untuk apa? Jelaskan input dan ouput dari Vaxijen!

A: analisis sekuens protein yang paling efisien sebagai antigen. Input 232 sekuens dalam .fas, output 1 sekuens dengan skor tertinggi. Threshold > 0.5

Reference Paper: https://doi.org/10.1186/1471-2105-8-4


In [126]:
with open("VaxiJen - predicting protective antigens.html") as f:
    read_data = f.read()
soup = BeautifulSoup(read_data, 'html.parser')
t = soup.find_all('font')
b = soup.find_all('b')
result = []
for i in range(int(len(t)/3)):
    try:
        y = 3 + (i*3)
        score = (b[y].contents)
        score = float(score[0].replace(' ',''))
        ID = t[y].contents
        ID = ID[0].replace('\n', '')
        seqx = t[y+1].contents
        seqfin = []
        for x in seqx:
            if type(x) == (bs4.element.NavigableString):
                seqfin.append(x)
        seq = "".join(seqfin)
        seq = str(seq.replace(' ',''))    
        data = [score, ID, seq]
        result.append(data)
    except IndexError as error:
        pass
#result[0]

In [21]:
df2 = pd.DataFrame(result, columns = ["Score", "ID", "Sequence"])
df2.sort_values(by=['Score'], ascending=False)

Score                                                 ID  \
78   0.6947  >tr|B4XPM1|B4XPM1_9FLAV Envelope glycoprotein ...   
114  0.6905  >tr|A0EVS1|A0EVS1_9FLAV Envelope glycoprotein ...   
55   0.6892  >tr|E9KIL7|E9KIL7_9FLAV Envelope glycoprotein ...   
229  0.6892  >tr|A0EVS4|A0EVS4_9FLAV Envelope glycoprotein ...   
157  0.6885  >tr|A0EVS3|A0EVS3_9FLAV Envelope glycoprotein ...   
231  0.6883  >tr|Q07693|Q07693_9FLAV Envelope glycoprotein ...   
152  0.6875  >tr|B4XPL7|B4XPL7_9FLAV Envelope glycoprotein ...   
225  0.6867  >tr|Q5TLA7|Q5TLA7_9FLAV Envelope glycoprotein ...   
33   0.6862  >tr|Q66367|Q66367_9FLAV Envelope glycoprotein ...   
43   0.6853  >tr|B4XPM0|B4XPM0_9FLAV Envelope glycoprotein ...   
204  0.6853  >tr|B4XPL4|B4XPL4_9FLAV Envelope glycoprotein ...   
59   0.6853  >tr|B4XPL6|B4XPL6_9FLAV Envelope glycoprotein ...   
38   0.6847  >tr|A0EVS2|A0EVS2_9FLAV Envelope glycoprotein ...   
50   0.6838  >tr|Q66369|Q66369_9FLAV Envelope glycoprotein ...   
181  0.6833  >tr|A0EVT5|A0EVT5_9FLAV Envelope glycoprotein ...   
224  0.6832  >tr|B4XPM2|B4XPM2_9FLAV Envelope glycoprotein ...   
41   0.6831  >tr|Q66368|Q66368_9FLAV Envelope glycoprotein ...   
76   0.6823  >tr|A8B6G8|A8B6G8_9FLAV Envelope glycoprotein ...   
145  0.6823  >tr|A8B6H0|A8B6H0_9FLAV Envelope glycoprotein ...   
113  0.6822  >tr|B4XPM6|B4XPM6_9FLAV Envelope glycoprotein ...   
92   0.6822  >tr|Q8JTK2|Q8JTK2_9FLAV Envelope glycoprotein ...   
87   0.6822  >tr|B4XPL5|B4XPL5_9FLAV Envelope glycoprotein ...   
3    0.6818  >tr|A0EVT7|A0EVT7_9FLAV Envelope glycoprotein ...   
22   0.6818  >tr|A0EVT8|A0EVT8_9FLAV Envelope glycoprotein ...   
23   0.6818  >tr|Q5TLA9|Q5TLA9_9FLAV Envelope glycoprotein ...   
19   0.6818  >tr|A0EVT9|A0EVT9_9FLAV Envelope glycoprotein ...   
199  0.6813  >tr|G4Y7A3|G4Y7A3_9FLAV Envelope glycoprotein ...   
48   0.6811  >tr|A8B6H1|A8B6H1_9FLAV Envelope glycoprotein ...   
139  0.6808  >tr|A0EVS6|A0EVS6_9FLAV Envelope glycoprotein ...   
172  0.6808  >tr|A5HMG7|A5HMG7_9FLAV Envelope glycoprotein ...   
..      ...                                                ...   
222  0.6381  >tr|Q6H1H4|Q6H1H4_9FLAV Envelope glycoprotein ...   
119  0.6379  >tr|Q6H1I5|Q6H1I5_9FLAV Envelope glycoprotein ...   
203  0.6370  >tr|Q6H1H6|Q6H1H6_9FLAV Envelope glycoprotein ...   
51   0.6369  >tr|Q6H1G7|Q6H1G7_9FLAV Envelope glycoprotein ...   
74   0.6369  >tr|Q6H1F8|Q6H1F8_9FLAV Envelope glycoprotein ...   
120  0.6369  >tr|Q6H1F6|Q6H1F6_9FLAV Envelope glycoprotein ...   
210  0.6369  >tr|Q6H1H2|Q6H1H2_9FLAV Envelope glycoprotein ...   
206  0.6369  >tr|Q6H1F4|Q6H1F4_9FLAV Envelope glycoprotein ...   
84   0.6369  >tr|Q6H1E7|Q6H1E7_9FLAV Envelope glycoprotein ...   
71   0.6369  >tr|Q6H1G3|Q6H1G3_9FLAV Envelope glycoprotein ...   
202  0.6367  >tr|Q6H1I1|Q6H1I1_9FLAV Envelope glycoprotein ...   
201  0.6364  >tr|D9J035|D9J035_9FLAV Envelope glycoprotein ...   
162  0.6362  >tr|Q6H1H8|Q6H1H8_9FLAV Envelope glycoprotein ...   
24   0.6362  >tr|Q6H1F3|Q6H1F3_9FLAV Envelope glycoprotein ...   
183  0.6362  >tr|Q6H1I2|Q6H1I2_9FLAV Envelope glycoprotein ...   
116  0.6362  >tr|Q6H1H9|Q6H1H9_9FLAV Envelope glycoprotein ...   
45   0.6360  >tr|Q6H1H1|Q6H1H1_9FLAV Envelope glycoprotein ...   
135  0.6360  >tr|Q6H1G9|Q6H1G9_9FLAV Envelope glycoprotein ...   
81   0.6358  >tr|Q6H1G1|Q6H1G1_9FLAV Envelope glycoprotein ...   
123  0.6357  >tr|Q6H1I8|Q6H1I8_9FLAV Envelope glycoprotein ...   
149  0.6357  >tr|Q6H1I3|Q6H1I3_9FLAV Envelope glycoprotein ...   
179  0.6347  >tr|Q6H1H5|Q6H1H5_9FLAV Envelope glycoprotein ...   
80   0.6347  >tr|Q6H1I9|Q6H1I9_9FLAV Envelope glycoprotein ...   
226  0.6347  >tr|Q6H1J0|Q6H1J0_9FLAV Envelope glycoprotein ...   
62   0.6344  >tr|B3RFU2|B3RFU2_9FLAV Envelope glycoprotein ...   
193  0.6334  >tr|Q6H1F1|Q6H1F1_9FLAV Envelope glycoprotein ...   
31   0.6289  >tr|Q6H1J5|Q6H1J5_9FLAV Envelope glycoprotein ...   
32   0.6244  >tr|Q9DXH7|Q9DXH7_9FLAV Envelope glycoprotein ...   
185  0.6195  >tr|Q5TLA6|Q5TLA6_9FLAV Envelope 

# Analyse NetCTL result
Q: NetCTL digunakan untuk apa? Jelaskan input dan output dari NetCTL!

A: prediksi epitop terbaik dari sekuens B4XPM1 (output vaxijen). Input: sekuens B4XPM1.fas. Output: combinatorial score terbaik dari masing-masing supertype (total 12) yang nantinya dipilih 5 terbaik. Threshold: 0.5, 0.89 dan 0.94 berturut-turut.

Reference Paper:https://doi.org/10.1186/1471-2105-8-424

In [127]:
with open("NetCTL-1.2 Server - prediction results.html") as f2:
    read_data2 = f2.read()
soup2 = BeautifulSoup(read_data2, 'html.parser')
net = soup2.find_all('pre')
net = net[0].contents
net = str(net[0])
net = net.split("\n")
net_result = net[2:-8]
net_hit = []
for i in net_result:
    data = i.split()
    if len(data) > 15:
        final_data = [int(data[0]), data[2], data[4], float(data[6]), float(data[8]), float(data[10]), float(data[12]), float(data[14])]
        net_hit.append(final_data)
#net_hit

In [128]:
df3 = pd.DataFrame(net_hit, columns = ["Res No", "ID", "sequence", "aff", "aff_rescale", "cle", "tap", "COMB"])
df3.sort_values(by=['COMB'], ascending=False)

Res No        ID   sequence     aff  aff_rescale     cle    tap    COMB
1     143  Sequence  ITEAELTGY  0.5917       2.5124  0.7508  2.836  2.7668
0     102  Sequence  VVQPENLEY  0.4255       1.8064  0.9463  3.194  2.1081
3     244  Sequence  QMSSGNLLF  0.2605       1.1059  0.7785  2.674  1.3564
2     173  Sequence  QMENKAWLV  0.2871       1.2189  0.6676  0.353  1.3366
6     445  Sequence  STSLSVSLV  0.2844       1.2077  0.3289  0.474  1.2807
7     453  Sequence  VLVGVVTLY  0.1681       0.7136  0.9726  3.132  1.0160
5     291  Sequence  GTIVIRVQY  0.1489       0.6322  0.9764  2.770  0.9172
4     264  Sequence  KLQLKGMSY  0.1217       0.5168  0.9684  2.880  0.8061